cmd/dp-import-tracker/main.go

```diff
@@ -0,0 +1,256 @@
+	}()
+
+	for {
+		select {
+		case newInstanceMsg := <-createInstanceChan:
+			if _, ok := trackedInstances[newInstanceMsg]; ok {
+				log.Error(errors.New("import instance exists"), log.Data{"instanceID": newInstanceMsg})
+			} else {
+				log.Debug("new import instance", log.Data{"instanceID": newInstanceMsg})
+				trackedInstances[newInstanceMsg] = trackedInstance{
+					totalObservations:         -1,
+					observationsInsertedCount: 0,
+				}
+			}
+		case updateObservationsInserted := <-updateInstanceWithObservationsInsertedChan:
+			instanceID := updateObservationsInserted.InstanceID
+			if _, ok := trackedInstances[instanceID]; !ok {
+				log.Info("Warning: import instance not in tracked list for update", log.Data{"update": updateObservationsInserted})
+			}
+			log.Debug("updating import instance", log.Data{"update": updateObservationsInserted})
+			if err := updateInstanceWithObservationsInserted(client, importAPIURL, instanceID, updateObservationsInserted.NumberOfObservationsInserted); err != nil {
+				log.ErrorC("failed to add inserts to instance", err, log.Data{"update": updateObservationsInserted})
+			}
+		case <-checkForCompletedInstancesChan:
+			log.Debug("check import Instances", log.Data{"q": trackedInstances})
+			for instanceID := range trackedInstances {
+				if err := trackedInstances.updateInstanceFromImportAPI(client, importAPIURL, instanceID); err != nil {
+					log.ErrorC("could not update instance", err, log.Data{"instanceID": instanceID})
+
+				} else if trackedInstances[instanceID].totalObservations > 0 && trackedInstances[instanceID].observationsInsertedCount >= trackedInstances[instanceID].totalObservations {
+					// the above check on totalObservations ensures that we have updated this instance from the Import API (non-default value)
+					// and that inserts have exceeded the expected
+					log.Debug("import instance complete", log.Data{"instanceID": instanceID, "observations": trackedInstances[instanceID].observationsInsertedCount})
+					log.Error(errors.New("TODO check db for actual count(insertedObservations) now that instance appears to be completed (kafka-double-counting?)"), nil)
+					if err := api.UpdateInstanceState(client, importAPIURL, instanceID, "completed"); err != nil {
+						log.ErrorC("Failed to set import instance state=completed", err, log.Data{"instanceID": instanceID, "observations": trackedInstances[instanceID].observationsInsertedCount})
+					} else if err := CheckImportJobCompletionState(client, importAPIURL, trackedInstances[instanceID].jobID, instanceID); err != nil {
+						log.ErrorC("Failed to check import job when instance completed", err, log.Data{"instanceID": instanceID, "jobID": trackedInstances[instanceID].jobID})
+					} else {
+						// no errors, so stop tracking the completed instance
+						delete(trackedInstances, instanceID)
+					}
+
+				}
+			}
+		}
+	}
+}
+
+func logFatal(context string, err error, data log.Data) {
+	log.ErrorC(context, err, data)
+	panic(err)
```

Comment:
I `panic` because this would give the stacktrace to allow the investigator to identify the line where `logFatal` was called - where things went wrong